Userland/Services/SpiceAgent/Message.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023, Caoimhe Byrne <caoimhebyrne06@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <AK/Vector.h>
+
+namespace SpiceAgent {
+
+static constexpr u32 AGENT_PROTOCOL = 1;
+
+// Used to communicate what the client/or server is capable of
+// TODO(Caoimhe): Document these types
+enum class Capability : u32 {
+    MouseState = 0,
+    MonitorsConfig,
+    Reply,
+    Clipboard,
+    DisplayConfig,
+    ClipboardByDemand,
+    ClipboardSelection,
+    SparseMonitorsConfig,
+    GuestLineEndLF,
+    GuestLineEndCRLF,
+    MaxClipboard,
+    AudioVolumeSync,
+    MonitorsConfigPosition,
+    FileTransferDisabled,
+    FileTransferDetailedErrors,
+    GraphicsCardInfo,
+    ClipboardNoReleaseOnRegrab,
+    ClipboardGrabSerial
+};
+
+class Message {
+public:
+    // TODO(Caoimhe): Document these types
+    enum class Type : u32 {
+        MouseState = 1,
+        MonitorsConfig,
+        Reply,
+        Clipboard,
+        DisplayConfig,
+        AnnounceCapabilities,
+        ClipboardGrab,
+        ClipboardRequest,
+        ClipboardRelease,
+        FileTransferStart,
+        FileTransferStatus,
+        FileTransferData,
+        Disconnected,
+        MaxClipboard,
+        VolumeSync,
+        GraphicsDeviceInfo
+    };
+
+    Message(Type type)
+        : m_type(type) {};
+
+    Type const& type() { return m_type; };
```

Comment:
Do we need to pass a reference here? you seem to pass a const reference to `m_type`, but if we don't need to change it (which is probably the case because you provide a const reference only), just pass it by value:
```c++
Type type() { return m_type; }
```